operator/pkg/lib/sidecars/restart/replica_set_action.go

```diff
@@ -0,0 +1,48 @@
+package restart
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getReplicaSetAction(ctx context.Context, c client.Client, pod v1.Pod, replicaSetRef *metav1.OwnerReference) (restartAction, error) {
+
+	replicaSetKey := client.ObjectKey{
+		Namespace: pod.Namespace,
+		Name:      replicaSetRef.Name,
+	}
+
+	var replicaSet = &appsv1.ReplicaSet{}
+	err := c.Get(ctx, replicaSetKey, replicaSet)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return newOwnerNotFoundAction(pod), nil
+		}
+		return restartAction{}, err
+	}
+
+	if rsOwnedBy, exists := getReplicaSetOwner(replicaSet); !exists {
+		return newDeleteAction(actionObjectFromPod(pod)), nil
```

Comment:
We could add a comment why in this case the pod will be deleted, to make it easier to understand without thinking about details.

Reply:
Discussion outcome: We should add a simple comment.